mule-extensions-api-xml-dsl/src/main/java/org/mule/runtime/extension/xml/dsl/api/resolver/DslElementResolver.java

```diff
@@ -349,25 +361,60 @@ private void declareFieldsAsChilds(final DslElementDeclarationBuilder objectBuil
 
     private boolean shouldGenerateChildElements(MetadataType metadataType, ExpressionSupport expressionSupport)
     {
-        boolean isExpressionRequired = ExpressionSupport.REQUIRED == expressionSupport;
-        boolean isPojo = metadataType instanceof ObjectType;
+        final AtomicBoolean supportsChildDeclaration = new AtomicBoolean(false);
 
-        Optional<ClassInformationAnnotation> classInformation = getSingleAnnotation(metadataType, ClassInformationAnnotation.class);
-        boolean isInstantiable = !isPojo || classInformation.map(ClassInformationAnnotation::isInstantiable).orElse(false);
-        boolean isExtensible = getSingleAnnotation(metadataType, ExtensibleTypeAnnotation.class).isPresent();
+        if (ExpressionSupport.REQUIRED == expressionSupport)
+        {
+            return false;
+        }
+
+        metadataType.accept(new MetadataTypeVisitor()
+        {
+            @Override
+            protected void defaultVisit(MetadataType metadataType)
+            {
+                supportsChildDeclaration.set(true);
+            }
+
+            @Override
+            public void visitArrayType(ArrayType arrayType)
+            {
+                arrayType.getType().accept(this);
+            }
+
+            @Override
+            public void visitObject(ObjectType objectType)
+            {
+                boolean isInstantiable = getSingleAnnotation(metadataType, ClassInformationAnnotation.class)
```

Comment:
we already have a Utils for this [here](https://github.com/mulesoft/mule/blob/mule-4.x/modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MetadataTypeUtils.java#L120), but since it is in `extensions-support` I don't have access to it from ext-api (neither to IntrospectionUtils).

Adding it to  `org.mule.metadata.utils.MetadataTypeUtils` present in `metadata-model-api` is not an option either since we cannot depend of `mule-metadata-model-java` which contains the `ClassInformationAnnotation`. 

Creating `isInstanciable` in `JavaTypeUtils` is an option that could work.